File: abc091/abc091_b.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
  int n,m;
  cin >> n;
  vector<string> s(n);
  for (int i = 0; i < n; i++)
  {
    cin >> s[i];
  }

  cin >> m;
  vector<string> t(m);
  for (int i = 0; i < m; i++)
  {
    cin >> t[i];
  }

  int ans = 0;
  int cnt = 0;
  for (int i = 0; i < n; i++)
  {
    for (int j = 0; j < n; j++)
    {
      if(s[i]==s[j]) cnt++;
    }
    for (int k = 0; k < m; k++)
    {
      if(s[i]==t[k]) cnt--;
    }
    ans = max(ans,cnt);
    cnt = 0;
  }
  
  cout << ans << endl;
}

File: abc091/abc091_b.rb
n=gets.to_i
s=[];
n.times do
  s << gets.chomp
end

m=gets.to_i
t=[];
m.times do
  t << gets.chomp
end

ans=0
cnt=0
s.each do |str1|
  s.each do |str2|
    if str1==str2
      cnt+=1
    end
  end
  t.each do |str3|
    if str1==str3
      cnt-=1
    end
  end
  ans = cnt if cnt>ans
  cnt = 0
end

puts ans